src/jdk.httpserver/share/classes/sun/net/httpserver/SSLStreams.java

```diff
@@ -85,8 +85,12 @@ private void configureEngine(HttpsConfigurator cfg, InetSocketAddress addr){
                         );
                     } catch (IllegalArgumentException e) { /* LOG */}
                 }
-                engine.setNeedClientAuth (params.getNeedClientAuth());
-                engine.setWantClientAuth (params.getWantClientAuth());
+                if (params.getNeedClientAuth()) {
+                    engine.setNeedClientAuth(true);
+                }
+                if (params.getWantClientAuth()) {
```

Comment:
Assume the states of `wantClientAuth` and `needClientAuth` are correctly maintained with the changes in `HttpsParameters`, or it's impossible both of them are `true`.
Could here use `if-else if` clause, like the below?
```
if (params.getNeedClientAuth()) {
    engine.setNeedClientAuth(true);
} else if (params.getWantClientAuth()) {
    engine.setWantClientAuth(true);
}
```